File: src/main/java/uk/gov/london/ops/service/project/LabelService.java
/**
 * Copyright (c) Greater London Authority, 2016.
 *
 * This source code is licensed under the Open Government Licence 3.0.
 *
 * http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
 */
package uk.gov.london.ops.service.project;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import uk.gov.london.ops.domain.project.Label;
import uk.gov.london.ops.repository.LabelRepository;

import java.util.Set;

@Service
/**
 * service for managing project / block labels
 */
public class LabelService {

    @Autowired
    private LabelRepository labelRepository;


    /**
     * creates a new label
     */
    public Label createLabel(Label label) {
        return labelRepository.save(label);
    }

    /**
     * Returns all labels used by specified block
     */
    public Set<Label> getLabelsForBlock(Integer blockId) {
        return labelRepository.findLabelsForBlock(blockId);
    }

    /**
     * Deletes a label by project Id
     */
    public void deleteLabelByProjectId(Integer projectId) {
        labelRepository.deleteByProjectId(projectId);
    }


}

File: gla-ops-application/src/main/ui/app/scripts/services/LabelService.js
/**
 * Copyright (c) Greater London Authority, 2016.
 *
 * This source code is licensed under the Open Government Licence 3.0.
 *
 * http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
 */

LabelService.$inject = ['$http', 'config'];

function LabelService($http, config) {
  return {
    getPreSetLabels: function (organisationId, markedForCorporateReport) {
      let cfg = {
        params: {
          managingOrganisationId: organisationId || null,
          markedForCorporate: markedForCorporateReport,
        }
      };

      return $http.get(`${config.basePath}/preSetLabels`, cfg);
    },

    createLabel(data) {
      return $http.post(`${config.basePath}/preSetLabels`, data);
    },

    updateLabel(data) {
      return $http.put(`${config.basePath}/preSetLabels/${data.id}`, data);
    },

    deleteLabel(data) {
      return $http.delete(`${config.basePath}/preSetLabels/${data.id}`);
    },

  }
}

angular.module('GLA')
  .service('LabelService', LabelService);
